### ISSUE TITLE
Store API - Visible Attributes are not returned in Product endpoints

### ISSUE THREAD

**Issue by JohnBelly95** [woocommerce/woocommerce-gutenberg-products-block#6180]
## Describe the bug

Simple Products do not return their visible attributes through the `/products` and `/products/:id` routes.

## To reproduce

1. Create a Simple Product with Attributes set to visible
2. Send a `GET` request to `/products`
3. Locate product in the response
4. See that the attributes field has returned an empty array

## Expected behavior

All Visible Attributes should be returned when retrieving Products, be it through listing or getting a single one.

## Screenshots

Screenshots are from a local deployment we created to inspect the issue more closely.

Admin Page showing that the Product has visible Attributes with values.
![Admin Page](https://user-images.githubusercontent.com/7394451/161291102-d22edd4d-f675-4fd3-9cf3-83f252476b97.png)

Same Product through Postman returning no attributes
![Postman Response](https://user-images.githubusercontent.com/7394451/161291733-c4b7aab0-7913-46c7-91b7-7c702abd1794.png)

### WordPress:

-   Core version: 5.9.2
-   WooCommerce version: 6.4.0-rc.1
-   Site language: English (UK)
-   Any other plugins installed: WooCommerce Beta Tester, WP API SwaggerUI

## Additional context

In the ProductSchema.php file in the source code, the description of the `attributes` field explains that it represents `List of attributes assigned to the product/variation that are visible or used for variations.` but around 400 lines beneath that, the `get_attributes()` function contains this code block:
```php
// Only visible and variation attributes will be exposed by this API.
if ( ! $attribute->get_visible() || ! $attribute->get_variation() ) {
	continue;
}
```

I think that either that `||` should be changed to `&&` to reflect the Schema description, or the `or` in the Schema description should be changed to `and` to reflect the functionality.

**Comment by whywilson** [comment]
Thank you for the nice explanation and solution. Look forward to getting this work done soon.

**Comment by senadir** [comment]
Should should be fixed in #6274 and out in WooCommerce blocks 7.5.0 (Due date April 30) and WooCommerce 6.6 (due a couple of months from now).
